lib/WorseReflection/Core/Inference/Resolver/ObjectCreationExpressionResolver.php

```diff
@@ -74,4 +76,20 @@ private function resolveClassType(NodeContextResolver $resolver, Frame $frame, O
         );
         return new GenericClassType($resolver->reflector(), $classType->name(), $templateMap->toArguments());
     }
+
+    private function resolveAnonymousClass(NodeContextResolver $resolver, Node $node): NodeContext
+    {
+        return NodeContextFactory::create(
+            'testing',
+            $node->getStartPosition(),
+            $node->getEndPosition(),
+            [
+                'symbol_type' => Symbol::CLASS_,
+                'type' => TypeFactory::fromStringWithReflector(
```

Comment:
why `fromString`?

Reply:
Copied that over from how the class definition resolver handles it.

Reply:
ok, well TypeFactory::reflectedClass` or whatever but this is where the `ReflectionClass` should be created I think, not in teh collection

Reply:
Yes using `Typefactory::reflectedClass` is better.